implementations/rust/ockam/ockam_command/src/secure_channel/listener/create.rs

```diff
@@ -33,21 +35,38 @@ pub struct SecureChannelListenerNodeOpts {
 }
 
 impl CreateCommand {
-    pub fn run(self, options: CommandGlobalOpts) {
-        let cfg = options.config;
-        let node = extract_address_value(&self.node_opts.at).unwrap_or_else(|_| "".to_string());
-        let port = cfg.get_node_port(&node).unwrap();
+    pub fn run(self, opts: CommandGlobalOpts) {
+        node_rpc(rpc, (opts, self));
+    }
+}
 
-        connect_to(port, self, |ctx, cmd, rte| async {
-            create_listener(&ctx, cmd.address, cmd.authorized_identifier, rte).await?;
-            drop(ctx);
-            Ok(())
-        });
+async fn rpc(ctx: Context, (opts, cmd): (CommandGlobalOpts, CreateCommand)) -> crate::Result<()> {
+    run_impl(&ctx, (opts, cmd)).await
+}
+
+async fn run_impl(
+    ctx: &Context,
+    (opts, cmd): (CommandGlobalOpts, CreateCommand),
+) -> crate::Result<()> {
+    let node = extract_address_value(&cmd.node_opts.at)?;
+    let mut rpc = Rpc::background(&ctx, &opts, &node)?;
+    let req = Request::post("/node/secure_channel_listener").body(
+        CreateSecureChannelListenerRequest::new(&cmd.address, cmd.authorized_identifiers),
+    );
+    rpc.request(req).await?;
+    if rpc.parse_response::<Vec<u8>>().is_ok() {
+        println!("/service/{}", cmd.address.address());
+        Ok(())
+    } else {
+        Err(crate::error::Error::new(
+            exitcode::CANTCREAT,
+            anyhow!("An error occurred while creating secure channel listener"),
+        ))
     }
 }
 
 pub async fn create_listener(
```

Comment:
reverting this removal since this function is still called [here](https://github.com/build-trust/ockam/blob/65a3e23b0785cff93f5ed5897d653d459da908af/implementations/rust/ockam/ockam_command/src/node/create.rs#L291) and I'm not sure how to better refactor this. any suggestion?

Reply:
We can worry about that in a separate PR. I'll create an issue for that :+1: